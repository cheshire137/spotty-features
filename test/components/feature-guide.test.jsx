import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import FeatureGuide from '../../src/components/feature-guide.jsx'

describe('FeatureGuide', () => {
  test('matches snapshot', () => {
    const component = <FeatureGuide activeView="trends" />
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('changes class based on active view', () => {
    let component = <FeatureGuide activeView="search" />
    expect(shallow(component).find('.title.search-view').length).toBe(1)

    component = <FeatureGuide activeView="trends" />
    expect(shallow(component).find('.title.search-view').length).toBe(0)
  })
})
