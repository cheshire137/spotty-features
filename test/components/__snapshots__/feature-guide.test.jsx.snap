exports[`FeatureGuide matches snapshot 1`] = `
<div>
  <h3
    className="title is-3 feature-guide-header">
    Audio features
  </h3>
  <dl
    className="feature-guide">
    <dt
      style={
        Object {
          "color": "#FF7EAB",
        }
      }>
      Acoustic
    </dt>
    <dd>
      A confidence measure of whether the track is acoustic.
    </dd>
    <dt
      style={
        Object {
          "color": "#0ABD9C",
        }
      }>
      Danceable
    </dt>
    <dd>
      Describes how suitable a track is for dancing based on tempo, rhythm stability, beat strength, and regularity.
    </dd>
    <dt
      style={
        Object {
          "color": "#e3b951",
        }
      }>
      Energetic
    </dt>
    <dd>
      A perceptual measure of intensity and activity. Energetic tracks are usually fast, loud, and noisy.
    </dd>
    <dt
      style={
        Object {
          "color": "#494949",
        }
      }>
      Instrumental
    </dt>
    <dd>
      Whether a track lacks vocals.
    </dd>
    <dt
      style={
        Object {
          "color": "#AF33C8",
        }
      }>
      Live
    </dt>
    <dd>
      Detects the presence of an audience.
    </dd>
    <dt>
      <span
        style={
          Object {
            "color": "#90B207",
          }
        }>
        Positive
      </span>
      <span>
         / 
      </span>
      <span
        style={
          Object {
            "color": "#004DD0",
          }
        }>
        Negative
      </span>
    </dt>
    <dd>
      Describes the musical positiveness of the track. High positivity sounds more happy, cheerful, or euphoric. High negativity sounds more sad, depressed, or angry.
    </dd>
    <dt
      style={
        Object {
          "color": "#f09945",
        }
      }>
      Speechy
    </dt>
    <dd>
      Detects the presence of spoken words. More than 66% is probably entirely spoken. Between 33% - 66% may contain both music and speech. Less than 33% most likely is music.
    </dd>
  </dl>
</div>
`;
